housekeeping-vacuum-tool/src/main/java/com/hotels/housekeeping/tool/vacuum/validate/TablesValidator.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.housekeeping.tool.vacuum.validate;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.thrift.TException;
+
+import com.hotels.housekeeping.tool.vacuum.conf.Table;
+import com.hotels.housekeeping.tool.vacuum.conf.TablesValidationConfig;
+
+public class TablesValidator {
+
+  private final TablesValidationConfig tableValidationConfig;
+
+  public TablesValidator(TablesValidationConfig tableValidationConfig) {
+    this.tableValidationConfig = tableValidationConfig;
+  }
+
+  /**
+   * @param metastore
+   * @param tables
+   * @return {@link ValidationResult}
+   */
+  public ValidationResult validate(IMetaStoreClient metastore, List<Table> tables) {
+    ValidationResult result = new ValidationResult();
+    for (Table table : tables) {
+      try {
+        org.apache.hadoop.hive.metastore.api.Table hiveTable = metastore
+            .getTable(table.getDatabaseName(), table.getTableName());
+        Map<String, String> parameters = getParameters(hiveTable.getParameters());
+        if (!allPropertiesExist(parameters)) {
+          result.addValidationFailure(new PropertyValidationFailure(table, tableValidationConfig, parameters));
+        }
+      } catch (TException e) {
+        result.addValidationFailure(new UnexpectedValidationFailure(table, e));
+      }
+    }
+    return result;
+  }
+
+  private Map<String, String> getParameters(Map<String, String> parameters) {
```

Comment:
getParameters -> extractParameters (I like avoiding the usage of "get" for things that aren't getters in the POJO sense)